app/models/host.rb

```diff
@@ -1346,22 +1346,17 @@ def first_cat_entry(name)
     Classification.first_cat_entry(name, self)
   end
 
-  # TODO: Rename this to scan_queue and rename scan_from_queue to scan to match
-  #   standard from other places.
-  def scan(userid = "system", _options = {})
+  def scan(userid = "system", options = {})
     log_target = "#{self.class.name} name: [#{name}], id: [#{id}]"
-
-    task = MiqTask.create(:name => "SmartState Analysis for '#{name}' ", :userid => userid)
-
     _log.info("Requesting scan of #{log_target}")
-    begin
-      MiqEvent.raise_evm_job_event(self, :type => "scan", :prefix => "request")
-    rescue => err
-      _log.warn("Error raising request scan event for #{log_target}: #{err.message}")
-      return
-    end
+    check_policy_prevent(:request_host_scan, :scan_queue, userid, options)
```

Comment:
@lfu We already have a `scan_from_queue` method on the `Host` model.  Can the logic here be reworked so the `check_policy_prevent` calls that method when the scan is not being stopped by policy?  Otherwise we are calling the `scan_queue` method just to put the scan on the queue.